src/pages/products/Products.js

```diff
@@ -1,49 +1,64 @@
-import React from "react";
-import { products } from "../../backend/db/products";
-import { filterByCategory, filterByPrice, filterByRating, sortBy } from "../../utils/filterUtils";
+import React, { useEffect } from "react";
+import axios from "axios";
+import {
+  filterByCategory,
+  filterByPrice,
+  filterByRating,
+  sortBy,
+} from "../../utils";
 import { FiltersPanel, ProductCard } from "../../components";
-import { useFilter } from "../../contexts/filterContext";
-
+import { useFilter } from "../../contexts";
 
 function Products() {
-  const {filterState} = useFilter()
-  const filteredByCatagory = filterByCategory(products, filterState.catagoryName);
-  const filteredByRating = filterByRating(filteredByCatagory, filterState.inputRating);
-  const filteredByPrice = filterByPrice(filteredByRating, filterState.inputPrice)
-  const sortedBy = sortBy(filteredByPrice, filterState.inputSort)
-  
+  const { filterState, filterDispatch } = useFilter();
+  useEffect(() => {
+    const getData = async () => {
+      await axios
+        .get("/api/products")
+        .then((response) => {
+          filterDispatch({
+            type: "INITIAL_PRODUCTS",
+            payload: response.data.products,
+          });
+        })
+        .catch((err) => {
+          console.log(err);
+        });
+    };
+    getData();
```

Comment:
Why not use IIFE instead ? it would be much easier

Reply:
Thanks for pointing out, I will learn about IIFE and implement in the project.